engine/apps/slack/scenarios/resolution_note.py

```diff
@@ -82,10 +83,11 @@ def process_scenario(
                 _slack_team_identity=slack_team_identity,
                 channel_id=channel_id,
             )
-        except KeyError:
-            self.open_warning_window(payload, warning_text)
-            return
-        except SlackMessage.DoesNotExist:
+        except (KeyError, SlackMessage.DoesNotExist):
```

Comment:
It's not clear for me, how it works. Why both KeyError and SlackMessage.DoesNotExist indicates that event belongs to another cluster?

Reply:
Getting a `KeyError` here means that this is not a thread message (thus can't be added as a resolution note). In this case we just return a 200 response to avoid multiple regions opening the same warning window multiple times.
Getting a `SlackMessage.DoesNotExist` means that there could be another region that can handle this event.

I changed it to two separate checks to make this clearer in https://github.com/grafana/oncall/pull/4518/commits/853a135ccc8e70dcb111e519be3d66ae0543216a